tldw_Server_API/app/core/Jobs/manager.py

```
@@ -1045,6 +1097,24 @@ def create_job(
             raise ValueError(f"Queue '{queue}' not allowed for domain '{domain}'. Allowed: {allowed_queues}")  # noqa: TRY003
 
+        # Fair-share scheduling: enforce per-user concurrency limits and adjust priority
+        if owner_user_id:
+            try:
+                scheduler = _get_fair_share()
+                active_count = self._count_active_jobs_for_user(owner_user_id)
+                if not scheduler.can_submit(int(owner_user_id), active_count):
+                    raise ValueError(  # noqa: TRY003
+                        f"User {owner_user_id} has reached the maximum concurrent job limit "
+                        f"({scheduler.max_per_user})"
+                    )
```

Ruff (0.15.6)

[warning] Abstract `raise` to an inner function

(TRY301)